### ISSUE TITLE
Help Getting Started with Task Memory for Custom Agent

### ISSUE THREAD

**Issue by RENEK-bool** [agentscope-ai/ReMe#58]
Hi! 👋

First of all, thank you for building ReMe! I'm particularly excited about the Task Memory feature, which enables agents to "learn from experience and perform better on similar tasks." I’d love to integrate this capability into my own agent to improve task accuracy and efficiency.

However, as someone relatively new to memory-augmented agents, I’m unsure how to get started with end-to-end integration of Task Memory, especially when building from scratch. I’d really appreciate your guidance on a few practical questions:

1. Is the [cookbook/bfcl](https://github.com/agentscope-ai/ReMe/tree/main/cookbook/bfcl) example the recommended reference for setting up a task memory pool from zero? Are there other tutorials, documentation pages, or end-to-end demo scripts that walk through the full workflow?
2. If I want to integrate Task Memory into my own agent, do I need to pre-build a dataset of task experiences and explicitly define success/failure criteria for tasks? Are there any other key considerations or best practices I should keep in mind during integration?

Thanks so much for your time and for open-sourcing this great toolkit. Looking forward to learning from your work! 😊

**Comment by zouyingcao** [comment]
Thanks for your interest in our ReMe!

1. Yes, you can refer to [doc/cookbook](https://github.com/agentscope-ai/ReMe/blob/main/docs/cookbook/bfcl/quickstart.md) for guidance. The main workflow is: 
collect agent trajectories on training data set (use_memory=False) -> start ReMe service and init the task memory pool -> run experiments on test set with task memory (use_memory=True).

2. We recommend collecting some trajectories from training set to construct the initial task memory pool. You can set `success_threshold` in [config/default.yaml](https://github.com/agentscope-ai/ReMe/blob/main/reme_ai/config/default.yaml) to different values according to different tasks, where trajectory.score<success_threshold represents task failure. 
```
summary_task_memory:
    flow_content: TrajectoryPreprocessOp(success_threshold=1.0) >> (SuccessExtractionOp()|FailureExtractionOp()|ComparativeExtractionOp(enable_soft_comparison=True)) >> MemoryValidationOp(validation_threshold=0.5) >> MemoryDeduplicationOp() >> UpdateVectorStoreOp()
```

Any question, please let us know! 💐

**Comment by RENEK-bool** [comment]
I ran into a question while trying to collect trajectories. In the paper, it states that 
N trajectories are sampled for each question q. I’m wondering if this N corresponds to the `num_trials` parameter in [run_bfcl.py](https://github.com/agentscope-ai/ReMe/blob/main/cookbook/bfcl/run_bfcl.py).

However, I noticed in [bfcl_agent.py](https://github.com/agentscope-ai/ReMe/blob/main/cookbook/bfcl/bfcl_agent.py) that if a task succeeds on the first attempt, the agent skips the remaining trials (i.e., if `num_trials=8` but the first run succeeds, only 1 trajectory is collected). 

This means the actual number of sampled trajectories per question is variable, rather than a fixed N. Is this behavior by design? I originally assumed that the agent would always generate exactly N trajectories per question—regardless of success—so I’d appreciate your clarification on the intended sampling strategy.
